### DISCUSSION TITLE
Publishing on testPypi

### DISCUSSION THREAD

**Discussion by PaulCarrascosa** [umyelab/LabGym#53] [General]
Hello ! 
Sorry to bother you with this kind of question but I've seen that you have done your own package on testPypi and on Pypi. I would like to do the same with my package but it doesn't work for me. When it should be published on testPypi after I used "twine upload --repository testpypi dist/*" and when I try to use the pip install they gave me on the site with "pip install -i https://test.pypi.org/simple/ testLWTools", I still have errors which probably come from my setup.py file but I don't know how to figure it out. If you have tips I'm totally open with that.

I can see that it worked for you so I came for your help.
Thanks for taking of your time.

Paul

**Comment by yujiahu415** [comment]
Hi Paul,

This discussion section is specific for LabGym-related questions. For more general questions, the discussion forums like StackOverflow (https://stackoverflow.com) might be more suitable for you to seek help.

But for your question, this blog may help you: https://medium.com/@joel.barmettler/how-to-upload-your-python-package-to-pypi-65edc5fe9c56, which lists all the steps to publish your own package on PyPi. Hope this can be helpful to you.
